### ISSUE TITLE
Fails to detect bugs

### ISSUE THREAD

**Issue by spokorski** [spotbugs/sonar-findbugs#274]
## Issue Description

I picked out a handful of rules in order to test that they actually detect what they say they do.  Specifically these five:

- findbugs:IL_INFINITE_LOOP
- findbugs:EC_NULL_ARG
- findbugs:IP_PARAMETER_IS_DEAD_BUT_OVERWRITTEN
- findbugs:INT_BAD_COMPARISON_WITH_INT_VALUE
- findbugs:SF_DEAD_STORE_DUE_TO_SWITCH_FALLTHROUGH

The code I put together is below.  When running an analysis, only 3 of these bugs are detected.  The infinite loop and dead parameter bugs are not detected.

All 5 rules are enabled in the quality profile, and are configured as major bugs.

## Environment

| Component | Version |
| ------------------ | ------- |
| SonarQube | 7.7 |
| Sonar-FindBugs | 3.10.0 |
| Java | 1.8.0_121 |

## Code

```java
public void infinite(Integer dead){
    // input param is overwritten
    dead = Integer.MAX_VALUE;

    if (dead.equals(null) || dead > Long.MAX_VALUE){
        // .equals() with null parameter
        // int/long comparision out of range
    }

    switch(dead){
        case 1:
            dead = 1;
        case 2:
            // overwrite due to fallthrough
            dead = 2;
            break;
    }

    while (true){
        // infinite loop with no exit condition
    }
}
```

## Results

![sq](https://user-images.githubusercontent.com/14926863/58643004-0a542b80-82c4-11e9-99e1-ac6404a70a7e.png)
